crates/bindings-csharp/Codegen/Module.cs

```diff
@@ -87,32 +102,57 @@ or SpecialType.System_Int64
             )
             && type.NullableAnnotation != NullableAnnotation.Annotated;
 
-        if (Attrs.HasFlag(ColumnAttrs.Unique) && !IsEquatable)
+        if (attrs.HasFlag(ColumnAttrs.Unique) && !IsEquatable)
         {
             throw new Exception(
                 $"{type} {Name} is not valid for Identity, PrimaryKey or PrimaryKeyAuto as it's not an equatable primitive."
             );
         }
     }
 
+    public ColumnAttrs GetAttrs(string tableName) => Attrs
```

Comment:
It looks like we only ever retrieve those arguments as a mask. Maybe better to store it as pre-aggregated mask grouped by table in the first place?

That would both make the stored array smaller, and those checks simpler.

Reply:
That is, you would have something like

```cs
field
.GetAttributes()
.GroupBy(attr => /* table name */, attr => /* mask */)
.Select(group => new TableAttrs(group.Key, attr.Aggregate(...)))
.ToImmutableArray()
```

and store that.

Reply:
Hmm, the grouping needs to be more complex tho because its not just by tableName, its tableName or null.

ie all attributes for the column regardless of what table handle is instantiated, masked into attributes for the column specific to the table handle being instantiated

At that point it becomes a tradeoff between memory and compute; there's a good case that recomputing data is either fast or just as fast as keeping it cached, when transforms are this simple and called this infrequently.

Reply:
> its not just by tableName, its tableName or null.

`null` is still fine as a key though, and it will allow to iterate over fewer items each time you call `GetAttrs` - and that happens quite a lot as it's used in couple of loops.

`GetAttrs` will stay either way, it's just that it will need to do less work for each individual column, so you both store less stuff in memory and spend less CPU time - not really a tradeoff.